### PR TITLE
qa: ignore disk quota exceeded failure in test

### PR DESCRIPTION
**Pull request by nmshelke**
Quota enforcement takes some time. It is documented.
Test should ignore 'Disk Quota Exceeded' failure.

Fixes: https://tracker.ceph.com/issues/56632
Signed-off-by: Nikhilkumar Shelke <nshelke@redhat.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by nmshelke** [comment]
jenkins test windows

File: qa/tasks/cephfs/test_volumes.py [hunk 1]
```diff
@@ -6518,7 +6518,11 @@ def test_subvolume_snapshot_clone_quota_exceeded(self):
         self._fs_cmd("subvolume", "create", self.volname, subvolume,"--mode=777", "--size", str(osize))
 
         # do IO, write 50 files of 1MB each to exceed quota. This mostly succeeds as quota enforcement takes time.
-        self._do_subvolume_io(subvolume, number_of_files=50)
+        try:
+            self._do_subvolume_io(subvolume, number_of_files=5000)
```
Comment (dparmar18) [inline qa/tasks/cephfs/test_volumes.py:6522]: Is there a need to increase `number_of_files` from `50` to `5000`?

Comment (nmshelke) [reply qa/tasks/cephfs/test_volumes.py:6522]: I increased it just for testing.

Comment (dparmar18) [reply qa/tasks/cephfs/test_volumes.py]: Oh, okay. Now it seems you have reduced it back to `50`?

File: qa/tasks/cephfs/test_volumes.py [hunk 2]
```diff
@@ -6518,7 +6518,11 @@ def test_subvolume_snapshot_clone_quota_exceeded(self):
         self._fs_cmd("subvolume", "create", self.volname, subvolume,"--mode=777", "--size", str(osize))
 
         # do IO, write 50 files of 1MB each to exceed quota. This mostly succeeds as quota enforcement takes time.
-        self._do_subvolume_io(subvolume, number_of_files=50)
+        try:
+            self._do_subvolume_io(subvolume, number_of_files=50)
+        except CommandFailedError:
+            # command failed due to Disk quota exceeded as quota enforcement takes some.
```
Comment (vshankar) [inline qa/tasks/cephfs/test_volumes.py:6524]: we'd need to check for errno being EDQUOT.

Comment (vshankar) [reply qa/tasks/cephfs/test_volumes.py:6524]: @nmshelke mentioned about the interface not returning appropriate error codes (returns 1 on failures).

Let's leave this as it is. Looks fine!

**Review by lxbsz — APPROVED** [review]

**Comment by nmshelke** [comment]
jenkins test windows

**Comment by nmshelke** [comment]
jenkins test make check arm64

**Review by vshankar — APPROVED** [review]

File: qa/tasks/cephfs/test_volumes.py (hunk 2 above)
Comment (kotreshhr) [reply qa/tasks/cephfs/test_volumes.py:6524]: @nmshelke Doesn't the following code work?

```
 except CommandFailedError as e:
     if e.exitstatus == EDQUOT:
         pass

```

**Review by kotreshhr — CHANGES_REQUESTED** [review]

File: qa/tasks/cephfs/test_volumes.py (hunk 2 above)
Comment (kotreshhr) [inline qa/tasks/cephfs/test_volumes.py:6524]: We should also change the comment here. As commented at line 6520, the testcase expects quota to be exceeded. So the current comment is confusing. We can just say

```
# ignore quota enforcement error.
```

**Review by dparmar18 — APPROVED** [review]

**Comment by vshankar** [comment]
Test running here: https://pulpito.ceph.com/vshankar-2022-09-05_05:44:13-fs-wip-vshankar-testing-20220901-100101-testing-default-smithi/

**Review by vshankar — APPROVED** [review]

**Comment by vshankar** [comment]
jenkins retest this please

**Pull request merged**